### ISSUE TITLE
k8s - "resilient" planes not very resilient

### ISSUE THREAD

**Issue by pulberg** [rancher/rancher#8457]
**Rancher Versions:**
Server: v1.5.3
healthcheck:
ipsec:
network-services:
scheduler:
kubernetes (if applicable): v1.5.4-rancher1-3

**Docker Version:**
1.12.6

**OS and where are the hosts located? (cloud, bare metal, etc):**
RHEL7.3, vmware

**Setup Details: (single node rancher vs. HA rancher, internal DB vs. external DB)**
single node Rancher, external mysqldb

**Environment Type: (Cattle/Kubernetes/Swarm/Mesos)**
k8s

**Steps to Reproduce:**
Not sure, there must have been some sort of network issue.  Regardless, the point of setting up k8s as "resilient separated planes" is to handle all kinds of failure scenarios and stay running.

**Results:** 
k8s dashboard does not render, tiller pod not up, dashboard pod not up, k8s and Rancher seem to be out of sync - 

kublet logs dashboard - 
```
4/9/2017 7:53:51 PME0410 00:53:51.779959   17592 pod_workers.go:184] Error syncing pod 13b14892-1a3c-11e7-ba9f-02b97a5c8527, skipping: failed to "StartContainer" for "kubernetes-dashboard" with CrashLoopBackOff: "Back-off 1m20s restarting failed container=kubernetes-dashboard pod=kubernetes-dashboard-2492700511-0mrd4_kube-system(13b14892-1a3c-11e7-ba9f-02b97a5c8527)"
4/9/2017 7:53:54 PMI0410 00:53:54.599244   17592 operation_executor.go:917] MountVolume.SetUp succeeded for volume "kubernetes.io/secret/dee39534-0220-11e7-8ff4-02b97a09c2ce-default-token-t2h9v" (spec.Name: "default-token-t2h9v") pod "dee39534-0220-11e7-8ff4-02b97a09c2ce" (UID: "dee39534-0220-11e7-8ff4-02b97a09c2ce").
4/9/2017 7:54:01 PME0410 00:54:01.848060   17592 kubelet_node_status.go:302] Error updating node status, will retry: Operation cannot be fulfilled on nodes "testp38wblsb.rjf.com": the object has been modified; please apply your changes to the latest version and try again
4/9/2017 7:54:03 PMI0410 00:54:03.647674   17592 operation_executor.go:917] MountVolume.SetUp succeeded for volume "kubernetes.io/secret/13b14892-1a3c-11e7-ba9f-02b97a5c8527-io-rancher-system-token-vw652" (spec.Name: "io-rancher-system-token-vw652") pod "13b14892-1a3c-11e7-ba9f-02b97a5c8527" (UID: "13b14892-1a3c-11e7-ba9f-02b97a5c8527").
4/9/2017 7:54:03 PMI0410 00:54:03.868239   17592 docker_manager.go:2519] checking backoff for container "kubernetes-dashboard" in pod "kubernetes-dashboard-2492700511-0mrd4"
4/9/2017 7:54:03 PMI0410 00:54:03.868523   17592 docker_manager.go:2533] Back-off 1m20s restarting failed container=kubernetes-dashboard pod=kubernetes-dashboard-2492700511-0mrd4_kube-system(13b14892-1a3c-11e7-ba9f-02b97a5c8527)
```

kubelet logs tiller - 
```
4/9/2017 7:59:21 PME0410 00:59:21.431066   16363 pod_workers.go:184] Error syncing pod 60b83710-1a3d-11e7-ba9f-02b97a5c8527, skipping: failed to "StartContainer" for "tiller" with CrashLoopBackOff: "Back-off 5m0s restarting failed container=tiller pod=tiller-deploy-3991468440-pktg9_kube-system(60b83710-1a3d-11e7-ba9f-02b97a5c8527)"
4/9/2017 7:59:21 PME0410 00:59:21.452970   16363 kubelet_node_status.go:302] Error updating node status, will retry: Operation cannot be fulfilled on nodes "testp26wblsb.rjf.com": the object has been modified; please apply your changes to the latest version and try again
4/9/2017 7:59:31 PME0410 00:59:31.678906   16363 kubelet_node_status.go:302] Error updating node status, will retry: Operation cannot be fulfilled on nodes "testp26wblsb.rjf.com": the object has been modified; please apply your changes to the latest version and try again
4/9/2017 7:59:33 PMI0410 00:59:33.232484   16363 operation_executor.go:917] MountVolume.SetUp succeeded for volume "kubernetes.io/secret/60b83710-1a3d-11e7-ba9f-02b97a5c8527-io-rancher-system-token-vw652" (spec.Name: "io-rancher-system-token-vw652") pod "60b83710-1a3d-11e7-ba9f-02b97a5c8527" (UID: "60b83710-1a3d-11e7-ba9f-02b97a5c8527").
4/9/2017 7:59:33 PMI0410 00:59:33.430434   16363 docker_manager.go:2519] checking backoff for container "tiller" in pod "tiller-deploy-3991468440-pktg9"
4/9/2017 7:59:33 PMI0410 00:59:33.430721   16363 docker_manager.go:2533] Back-off 5m0s restarting failed container=tiller pod=tiller-deploy-3991468440-pktg9_kube-system(60b83710-1a3d-11e7-ba9f-02b97a5c8527)
```

I also have shutdown docker on a specific node **testp29wblsb**, yet "kubectl get nodes" still shows it as "ready"

```
[devopsadmin@testp29wblsb ~]$ sudo systemctl status docker.service
● docker.service - Docker Application Container Engine
   Loaded: loaded (/usr/lib/systemd/system/docker.service; enabled; vendor preset: disabled)
   Active: inactive (dead) since Sun 2017-04-09 20:40:15 EDT; 16min ago
```
```
➜ kubectl get nodes
NAME                   STATUS    AGE
testp26wblsb.rjf.com   Ready     35d
testp27wblsb.rjf.com   Ready     35d
testp28wblsb.rjf.com   Ready     35d
testp29wblsb.rjf.com   Ready     35d
testp38wblsb.rjf.com   Ready     34d
testp41wblsb.rjf.com   Ready     34d
```

**Comment by joshwget** [comment]
There are some general networking bug fixes coming in 1.5.4. If we're able to find a path to reproduce this issue we can check whether these will be solved in 1.5.4 and fix them if not. If you have any other details on how you might have gotten into this state it would be very helpful.

When you shutdown Docker, were its containers still running? If Kubelet is still running and active this is the behavior I'd expect. If not it's pretty surprising to me that Kubernetes wouldn't detect that the connection to that node was broken.

**Comment by pulberg** [comment]
*Update after upgrade to v1.5.4, all services were up and running for awhile, then got alerts for `r-kubernetes-scheduler-1-ecc3ad39` and `r-kubernetes-controller-manager-1-8a6f8616` restarting - 

scheduler - 
```
4/11/2017 12:17:47 AMI0411 05:17:47.217467       1 leaderelection.go:247] lock is held by 767261f2d3e4 and has not yet expired
4/11/2017 12:17:49 AMI0411 05:17:49.217929       1 leaderelection.go:247] lock is held by 767261f2d3e4 and has not yet expired
4/11/2017 12:17:51 AMI0411 05:17:51.218136       1 leaderelection.go:247] lock is held by 767261f2d3e4 and has not yet expired
4/11/2017 12:17:53 AMI0411 05:17:53.217453       1 leaderelection.go:247] lock is held by 767261f2d3e4 and has not yet expired
4/11/2017 12:17:55 AMI0411 05:17:55.232251       1 leaderelection.go:247] lock is held by 767261f2d3e4 and has not yet expired
4/11/2017 12:17:55 AMI0411 05:17:55.247737       1 leaderelection.go:247] lock is held by 767261f2d3e4 and has not yet expired
4/11/2017 12:17:55 AMI0411 05:17:55.247790       1 leaderelection.go:207] failed to renew lease kube-system/kube-scheduler
4/11/2017 12:17:55 AMF0411 05:17:55.247808       1 server.go:169] lost master
4/11/2017 12:17:55 AM88d15366-de6f-438b-abd9-741d136c933c
4/11/2017 12:18:17 AM+ '[' kube-scheduler == kubelet ']'
4/11/2017 12:18:17 AM+ curl -s -f http://rancher-metadata/2015-12-19/stacks/Kubernetes/services/kubernetes/uuid
4/11/2017 12:18:17 AM+ /usr/bin/update-rancher-ssl
4/11/2017 12:18:17 AM++ curl -s http://rancher-metadata/2015-12-19/stacks/Kubernetes/services/kubernetes/uuid
4/11/2017 12:18:17 AM+ UUID=88d15366-de6f-438b-abd9-741d136c933c
4/11/2017 12:18:17 AM++ curl -s -u : '/services?uuid=88d15366-de6f-438b-abd9-741d136c933c'
4/11/2017 12:18:17 AM++ jq -r '.data[0].actions.certificate'
4/11/2017 12:18:17 AM+ ACTION=
4/11/2017 12:18:17 AM+ KUBERNETES_URL=https://kubernetes:6443
4/11/2017 12:18:17 AM+ '[' -n '' ']'
4/11/2017 12:18:17 AM+ '[' kube-scheduler == kubelet ']'
4/11/2017 12:18:17 AM+ '[' kube-scheduler == kube-apiserver ']'
4/11/2017 12:18:17 AM+ exec kube-scheduler --master=http://kubernetes.kubernetes.rancher.internal --address=0.0.0.0
4/11/2017 12:18:17 AMI0411 05:18:17.235850       1 leaderelection.go:247] lock is held by 767261f2d3e4 and has not yet expired
4/11/2017 12:18:20 AMI0411 05:18:20.290063       1 leaderelection.go:247] lock is held by 767261f2d3e4 and has not yet expired
4/11/2017 12:18:23 AMI0411 05:18:23.208241       1 leaderelection.go:247] lock is held by 767261f2d3e4 and has not yet expired
4/11/2017 12:18:25 AMI0411 05:18:25.976988       1 leaderelection.go:247] lock is held by 767261f2d3e4 and has not yet expired
4/11/2017 12:18:29 AMI0411 05:18:29.106065       1 leaderelection.go:247] lock is held by 767261f2d3e4 and has not yet expired
4/11/2017 12:18:31 AMI0411 05:18:31.789473       1 leaderelection.go:247] lock is held by 767261f2d3e4 and has not yet expired
4/11/2017 12:18:49 AME0411 05:18:49.504307       1 leaderelection.go:261] Failed to update lock: client: etcd cluster is unavailable or misconfigured
4/11/2017 12:19:08 AME0411 05:19:08.142811       1 leaderelection.go:261] Failed to update lock: client: etcd cluster is unavailable or misconfigured
4/11/2017 12:19:25 AME0411 05:19:25.680062       1 leaderelection.go:261] Failed to update lock: client: etcd cluster is unavailable or misconfigured
4/11/2017 12:19:43 AME0411 05:19:43.181055       1 leaderelection.go:261] Failed to update lock: client: etcd cluster is unavailable or misconfigured
4/11/2017 12:20:01 AME0411 05:20:01.059517       1 leaderelection.go:261] Failed to update lock: client: etcd cluster is unavailable or misconfigured
4/11/2017 12:20:19 AME0411 05:20:19.441087       1 leaderelection.go:261] Failed to update lock: client: etcd cluster is unavailable or misconfigured
4/11/2017 12:20:38 AME0411 05:20:38.524496       1 leaderelection.go:261] Failed to update lock: client: etcd cluster is unavailable or misconfigured
4/11/2017 12:20:56 AME0411 05:20:56.240687       1 leaderelection.go:261] Failed to update lock: client: etcd cluster is unavailable or misconfigured
```
output of `etcdctl cluster-helath` shows 1 node "unhealthy", but that node appears fine in the UI - 
```
bash-4.3# etcdctl cluster-health
member 5b3f4aace201 is healthy: got healthy result from http://10.42.100.81:2379
member dcf0ef0b7b1275fe is healthy: got healthy result from http://10.42.58.246:
2379
member ff33ba045c3a1c84 is unhealthy: got unhealthy result from http://10.42.13.
23:2379
cluster is healthy
```
![screen shot 2017-04-11 at 12 33 50 am](https://cloud.githubusercontent.com/assets/11069351/24894231/97d9fa2a-1e4e-11e7-987a-aba54e152aef.png)

logs on "unhealthy" node - 
```
4/11/2017 12:35:05 AM2017-04-11 05:35:05.618578 I | raft: 30c19612dc90d9d is starting a new election at term 2279
4/11/2017 12:35:05 AM2017-04-11 05:35:05.618618 I | raft: 30c19612dc90d9d became candidate at term 2280
4/11/2017 12:35:05 AM2017-04-11 05:35:05.618627 I | raft: 30c19612dc90d9d received vote from 30c19612dc90d9d at term 2280
4/11/2017 12:35:05 AM2017-04-11 05:35:05.618654 I | raft: 30c19612dc90d9d [logterm: 18, index: 1010912] sent vote request to ad12df73ee6f967 at term 2280
4/11/2017 12:35:05 AM2017-04-11 05:35:05.621590 E | rafthttp: request sent was ignored (cluster ID mismatch: remote[ad12df73ee6f967]=5b3f4aace202, local=cc5589e5cbdc5204)
4/11/2017 12:35:05 AM2017-04-11 05:35:05.687239 E | rafthttp: request cluster ID mismatch (got 5b3f4aace202 want cc5589e5cbdc5204)
4/11/2017 12:35:05 AM2017-04-11 05:35:05.687360 E | rafthttp: request sent was ignored (cluster ID mismatch: remote[ad12df73ee6f967]=5b3f4aace202, local=cc5589e5cbdc5204)
4/11/2017 12:35:05 AM2017-04-11 05:35:05.687410 E | rafthttp: request sent was ignored (cluster ID mismatch: remote[ad12df73ee6f967]=5b3f4aace202, local=cc5589e5cbdc5204)
4/11/2017 12:35:05 AM2017-04-11 05:35:05.687500 E | rafthttp: request cluster ID mismatch (got 5b3f4aace202 want cc5589e5cbdc5204)
4/11/2017 12:35:05 AM2017-04-11 05:35:05.688998 E | rafthttp: request cluster ID mismatch (got 5b3f4aace202 want cc5589e5cbdc5204)
4/11/2017 12:35:05 AM2017-04-11 05:35:05.689199 E | rafthttp: request cluster ID mismatch (got 5b3f4aace202 want cc5589e5cbdc5204)
4/11/2017 12:35:05 AM2017-04-11 05:35:05.791396 E | rafthttp: request cluster ID mismatch (got 5b3f4aace202 want cc5589e5cbdc5204)
4/11/2017 12:35:05 AM2017-04-11 05:35:05.791602 E | rafthttp: request cluster ID mismatch (got 5b3f4aace202 want cc5589e5cbdc5204)
4/11/2017 12:35:05 AM2017-04-11 05:35:05.792442 E | rafthttp: request sent was ignored (cluster ID mismatch: remote[ad12df73ee6f967]=5b3f4aace202, local=cc5589e5cbdc5204)
4/11/2017 12:35:05 AM2017-04-11 05:35:05.792645 E | rafthttp: request sent was ignored (cluster ID mismatch: remote[ad12df73ee6f967]=5b3f4aace202, local=cc5589e5cbdc5204)
4/11/2017 12:35:05 AM2017-04-11 05:35:05.793355 E | rafthttp: request cluster ID mismatch (got 5b3f4aace202 want cc5589e5cbdc5204)
4/11/2017 12:35:05 AM2017-04-11 05:35:05.793526 E | rafthttp: request cluster ID mismatch (got 5b3f4aace202 want cc5589e5cbdc5204)
4/11/2017 12:35:05 AM2017-04-11 05:35:05.895733 E | rafthttp: request cluster ID mismatch (got 5b3f4aace202 want cc5589e5cbdc5204)
4/11/2017 12:35:05 AM2017-04-11 05:35:05.895887 E | rafthttp: request cluster ID mismatch (got 5b3f4aace202 want cc5589e5cbdc5204)
4/11/2017 12:35:05 AM2017-04-11 05:35:05.896800 E | rafthttp: request sent was ignored (cluster ID mismatch: remote[ad12df73ee6f967]=5b3f4aace202, local=cc5589e5cbdc5204)
4/11/2017 12:35:05 AM2017-04-11 05:35:05.896921 E | rafthttp: request sent was ignored (cluster ID mismatch: remote[ad12df73ee6f967]=5b3f4aace202, local=cc5589e5cbdc5204)
4/11/2017 12:35:05 AM2017-04-11 05:35:05.897397 E | rafthttp: request cluster ID mismatch (got 5b3f4aace202 want cc5589e5cbdc5204)
4/11/2017 12:35:05 AM2017-04-11 05:35:05.897513 E | rafthttp: request cluster ID mismatch (got 5b3f4aace202 want cc5589e5cbdc5204)
4/11/2017 12:35:05 AM2017-04-11 05:35:05.999208 E | rafthttp: request cluster ID mismatch (got 5b3f4aace202 want cc5589e5cbdc5204)
4/11/2017 12:35:05 AM2017-04-11 05:35:05.999266 E | rafthttp: request cluster ID mismatch (got 5b3f4aace202 want cc5589e5cbdc5204)
4/11/2017 12:35:06 AM2017-04-11 05:35:06.000687 E | rafthttp: request cluster ID mismatch (got 5b3f4aace202 want cc5589e5cbdc5204)
4/11/2017 12:35:06 AM2017-04-11 05:35:06.000975 E | rafthttp: request sent was ignored (cluster ID mismatch: remote[ad12df73ee6f967]=5b3f4aace202, local=cc5589e5cbdc5204)
4/11/2017 12:35:06 AM2017-04-11 05:35:06.001098 E | rafthttp: request sent was ignored (cluster ID mismatch: remote[ad12df73ee6f967]=5b3f4aace202, local=cc5589e5cbdc5204)
4/11/2017 12:35:06 AM2017-04-11 05:35:06.002730 E | rafthttp: request cluster ID mismatch (got 5b3f4aace202 want cc5589e5cbdc5204)
4/11/2017 12:35:06 AM2017-04-11 05:35:06.005712 E | rafthttp: request cluster ID mismatch (got 5b3f4aace202 want cc5589e5cbdc5204)
4/11/2017 12:35:06 AM2017-04-11 05:35:06.103851 E | rafthttp: request cluster ID mismatch (got 5b3f4aace202 want cc5589e5cbdc5204)
4/11/2017 12:35:06 AM2017-04-11 05:35:06.104095 E | rafthttp: request cluster ID mismatch (got 5b3f4aace202 want cc5589e5cbdc5204)
4/11/2017 12:35:06 AM2017-04-11 05:35:06.104661 E | rafthttp: request cluster ID mismatch (got 5b3f4aace202 want cc5589e5cbdc5204)
```

logs from "controller-manager" - 
```
4/11/2017 12:17:46 AMI0411 05:17:46.485121       1 leaderelection.go:247] lock is held by 55611939f2fc and has not yet expired
4/11/2017 12:17:48 AMI0411 05:17:48.484396       1 leaderelection.go:247] lock is held by 55611939f2fc and has not yet expired
4/11/2017 12:17:50 AMI0411 05:17:50.484038       1 leaderelection.go:247] lock is held by 55611939f2fc and has not yet expired
4/11/2017 12:17:52 AMI0411 05:17:52.484852       1 leaderelection.go:247] lock is held by 55611939f2fc and has not yet expired
4/11/2017 12:17:54 AMI0411 05:17:54.484008       1 leaderelection.go:247] lock is held by 55611939f2fc and has not yet expired
4/11/2017 12:17:54 AMI0411 05:17:54.489811       1 leaderelection.go:247] lock is held by 55611939f2fc and has not yet expired
4/11/2017 12:17:54 AMI0411 05:17:54.489851       1 leaderelection.go:207] failed to renew lease kube-system/kube-controller-manager
4/11/2017 12:17:54 AMF0411 05:17:54.489977       1 controllermanager.go:216] leaderelection lost
4/11/2017 12:18:02 AM+ '[' kube-controller-manager == kubelet ']'
4/11/2017 12:18:02 AM+ curl -s -f http://rancher-metadata/2015-12-19/stacks/Kubernetes/services/kubernetes/uuid
4/11/2017 12:18:02 AM+ /usr/bin/update-rancher-ssl
4/11/2017 12:18:02 AM++ curl -s http://rancher-metadata/2015-12-19/stacks/Kubernetes/services/kubernetes/uuid
4/11/2017 12:18:02 AM+ UUID=88d15366-de6f-438b-abd9-741d136c933c
4/11/2017 12:18:02 AM++ curl -s -u F76DFDA14792B2376C2F:njzFLRgMv2j2QpAQmJbY3iMf9Uo6UYQFZ8YJVtXV 'http://testp04wblsb.rjf.com:8080/v1/services?uuid=88d15366-de6f-438b-abd9-741d136c933c'
4/11/2017 12:18:02 AM++ jq -r '.data[0].actions.certificate'
4/11/2017 12:18:02 AM+ ACTION='http://testp04wblsb.rjf.com:8080/v1/services/1s46/?action=certificate'
4/11/2017 12:18:02 AM+ KUBERNETES_URL=https://kubernetes:6443
4/11/2017 12:18:02 AM+ '[' -n 'http://testp04wblsb.rjf.com:8080/v1/services/1s46/?action=certificate' ']'
4/11/2017 12:18:02 AM+ mkdir -p /etc/kubernetes/ssl
4/11/2017 12:18:02 AM+ cd /etc/kubernetes/ssl
4/11/2017 12:18:02 AM+ curl -s -u F76DFDA14792B2376C2F:njzFLRgMv2j2QpAQmJbY3iMf9Uo6UYQFZ8YJVtXV -X POST 'http://testp04wblsb.rjf.com:8080/v1/services/1s46/?action=certificate'
4/11/2017 12:18:02 AM+ unzip -o certs.zip
4/11/2017 12:18:02 AM88d15366-de6f-438b-abd9-741d136c933cArchive:  certs.zip
4/11/2017 12:18:02 AM  inflating: ca.pem
4/11/2017 12:18:02 AM  inflating: cert.pem
4/11/2017 12:18:02 AM  inflating: key.pem
4/11/2017 12:18:02 AM+ cd /
4/11/2017 12:18:02 AM+ cat
4/11/2017 12:18:02 AM+ '[' kube-controller-manager == kubelet ']'
4/11/2017 12:18:02 AM+ '[' kube-controller-manager == kube-apiserver ']'
4/11/2017 12:18:02 AM+ exec kube-controller-manager --master=https://kubernetes.kubernetes.rancher.internal:6443 --cloud-provider=rancher --address=0.0.0.0 --kubeconfig=/etc/kubernetes/ssl/kubeconfig --root-ca-file=/etc/kubernetes/ssl/ca.pem --service-account-private-key-file=/etc/kubernetes/ssl/key.pem
4/11/2017 12:18:02 AMI0411 05:18:02.471185       1 leaderelection.go:247] lock is held by 55611939f2fc and has not yet expired
4/11/2017 12:18:05 AMI0411 05:18:05.941440       1 leaderelection.go:247] lock is held by 55611939f2fc and has not yet expired
4/11/2017 12:18:08 AMI0411 05:18:08.998944       1 leaderelection.go:247] lock is held by 55611939f2fc and has not yet expired
4/11/2017 12:18:12 AMI0411 05:18:12.024755       1 leaderelection.go:247] lock is held by 55611939f2fc and has not yet expired
4/11/2017 12:18:15 AMI0411 05:18:15.678136       1 leaderelection.go:247] lock is held by 55611939f2fc and has not yet expired
4/11/2017 12:18:32 AME0411 05:18:32.858423       1 leaderelection.go:261] Failed to update lock: client: etcd cluster is unavailable or misconfigured
4/11/2017 12:18:50 AME0411 05:18:50.255639       1 leaderelection.go:261] Failed to update lock: client: etcd cluster is unavailable or misconfigured
```

**Comment by pulberg** [comment]
*Update 4/11 - 

Got restart alerts for `r-kubernetes-rancher-kubernetes-agent-1-4ee6fa77`, `r-kubernetes-kubectld-1-0f329a60`

kubernetes-agent logs - 
```
4/11/2017 8:25:11 AMRequest => GET http://kubernetes.kubernetes.rancher.internal/api/v1/namespaces/default
4/11/2017 8:25:11 AMResponse <= {"kind":"Namespace","apiVersion":"v1","metadata":{"name":"default","selfLink":"/api/v1/namespacesdefault","uid":"7b25ea1f-0170-11e7-9b39-02b97a15d24b","resourceVersion":"14","creationTimestamp":"2017-03-05T06:53:47Z"},"spec":{"finalizers":["kubernetes"]},"status":{"phase":"Active"}}
4/11/2017 8:25:11 AM
4/11/2017 8:25:16 AMRequest => GET http://kubernetes.kubernetes.rancher.internal/api/v1/nodes/testp08wblsb.rjf.com
4/11/2017 8:25:16 AMtime="2017-04-11T13:25:16Z" level=error msg="Error getting node: [testp08wblsb.rjf.com] by name from kubernetes, err: [Bad response from [http://kubernetes.kubernetes.rancher.internal/api/v1/nodes/testp08wblsb.rjf.com], statusCode [404]. Status [404 Not Found]. Body: [{\"kind\":\"Status\",\"apiVersion\":\"v1\",\"metadata\":{},\"status\":\"Failure\",\"message\":\"nodes \\\"testp08wblsb.rjf.com\\\" not found\",\"reason\":\"NotFound\",\"details\":{\"name\":\"testp08wblsb.rjf.com\",\"kind\":\"nodes\"},\"code\":404}\n]]"
4/11/2017 8:25:16 AMtime="2017-04-11T13:25:16Z" level=error msg="Error getting node: [testp08wblsb.rjf.com] by name from kubernetes, err: [Bad response from [http://kubernetes.kubernetes.rancher.internal/api/v1/nodes/testp08wblsb.rjf.com], statusCode [404]. Status [404 Not Found]. Body: [{\"kind\":\"Status\",\"apiVersion\":\"v1\",\"metadata\":{},\"status\":\"Failure\",\"message\":\"nodes \\\"testp08wblsb.rjf.com\\\" not found\",\"reason\":\"NotFound\",\"details\":{\"name\":\"testp08wblsb.rjf.com\",\"kind\":\"nodes\"},\"code\":404}\n]]"
4/11/2017 8:25:16 AMRequest => GET http://kubernetes.kubernetes.rancher.internal/api/v1/nodes/testp33wblsb.rjf.com
4/11/2017 8:25:16 AMRequest => GET http://kubernetes.kubernetes.rancher.internal/api/v1/nodes/testp36wblsb.rjf.com
4/11/2017 8:25:16 AMtime="2017-04-11T13:25:16Z" level=error msg="Error getting node: [testp33wblsb.rjf.com] by name from kubernetes, err: [Bad response from [http://kubernetes.kubernetes.rancher.internal/api/v1/nodes/testp33wblsb.rjf.com], statusCode [404]. Status [404 Not Found]. Body: [{\"kind\":\"Status\",\"apiVersion\":\"v1\",\"metadata\":{},\"status\":\"Failure\",\"message\":\"nodes \\\"testp33wblsb.rjf.com\\\" not found\",\"reason\":\"NotFound\",\"details\":{\"name\":\"testp33wblsb.rjf.com\",\"kind\":\"nodes\"},\"code\":404}\n]]"
4/11/2017 8:25:16 AMtime="2017-04-11T13:25:16Z" level=error msg="Error getting node: [testp33wblsb.rjf.com] by name from kubernetes, err: [Bad response from [http://kubernetes.kubernetes.rancher.internal/api/v1/nodes/testp33wblsb.rjf.com], statusCode [404]. Status [404 Not Found]. Body: [{\"kind\":\"Status\",\"apiVersion\":\"v1\",\"metadata\":{},\"status\":\"Failure\",\"message\":\"nodes \\\"testp33wblsb.rjf.com\\\" not found\",\"reason\":\"NotFound\",\"details\":{\"name\":\"testp33wblsb.rjf.com\",\"kind\":\"nodes\"},\"code\":404}\n]]"
4/11/2017 8:25:16 AMRequest => GET http://kubernetes.kubernetes.rancher.internal/api/v1/nodes/testp16wblsb.rjf.com
4/11/2017 8:25:16 AMtime="2017-04-11T13:25:16Z" level=error msg="Error getting node: [testp36wblsb.rjf.com] by name from kubernetes, err: [Bad response from [http://kubernetes.kubernetes.rancher.internal/api/v1/nodes/testp36wblsb.rjf.com], statusCode [404]. Status [404 Not Found]. Body: [{\"kind\":\"Status\",\"apiVersion\":\"v1\",\"metadata\":{},\"status\":\"Failure\",\"message\":\"nodes \\\"testp36wblsb.rjf.com\\\" not found\",\"reason\":\"NotFound\",\"details\":{\"name\":\"testp36wblsb.rjf.com\",\"kind\":\"nodes\"},\"code\":404}\n]]"
4/11/2017 8:25:16 AMtime="2017-04-11T13:25:16Z" level=error msg="Error getting node: [testp36wblsb.rjf.com] by name from kubernetes, err: [Bad response from [http://kubernetes.kubernetes.rancher.internal/api/v1/nodes/testp36wblsb.rjf.com], statusCode [404]. Status [404 Not Found]. Body: [{\"kind\":\"Status\",\"apiVersion\":\"v1\",\"metadata\":{},\"status\":\"Failure\",\"message\":\"nodes \\\"testp36wblsb.rjf.com\\\" not found\",\"reason\":\"NotFound\",\"details\":{\"name\":\"testp36wblsb.rjf.com\",\"kind\":\"nodes\"},\"code\":404}\n]]"
4/11/2017 8:25:16 AMtime="2017-04-11T13:25:16Z" level=error msg="Error getting node: [testp16wblsb.rjf.com] by name from kubernetes, err: [Bad response from [http://kubernetes.kubernetes.rancher.internal/api/v1/nodes/testp16wblsb.rjf.com], statusCode [404]. Status [404 Not Found]. Body: [{\"kind\":\"Status\",\"apiVersion\":\"v1\",\"metadata\":{},\"status\":\"Failure\",\"message\":\"nodes \\\"testp16wblsb.rjf.com\\\" not found\",\"reason\":\"NotFound\",\"details\":{\"name\":\"testp16wblsb.rjf.com\",\"kind\":\"nodes\"},\"code\":404}\n]]"
4/11/2017 8:25:16 AMtime="2017-04-11T13:25:16Z" level=error msg="Error getting node: [testp16wblsb.rjf.com] by name from kubernetes, err: [Bad response from [http://kubernetes.kubernetes.rancher.internal/api/v1/nodes/testp16wblsb.rjf.com], statusCode [404]. Status [404 Not Found]. Body: [{\"kind\":\"Status\",\"apiVersion\":\"v1\",\"metadata\":{},\"status\":\"Failure\",\"message\":\"nodes \\\"testp16wblsb.rjf.com\\\" not found\",\"reason\":\"NotFound\",\"details\":{\"name\":\"testp16wblsb.rjf.com\",\"kind\":\"nodes\"},\"code\":404}\n]]"
```

No good logs for kubectld, but the kubernetes stack has some - 
```
4/11/2017 8:38:59 AME0411 13:38:59.682920       1 repair.go:149] the cluster IP 10.43.215.251 for service heapster/kube-system is not allocated; repairing
4/11/2017 8:38:59 AME0411 13:38:59.682990       1 repair.go:149] the cluster IP 10.43.46.245 for service kubernetes-dashboard/kube-system is not allocated; repairing
4/11/2017 8:38:59 AME0411 13:38:59.684161       1 repair.go:149] the cluster IP 10.43.176.229 for service monitoring-grafana/kube-system is not allocated; repairing
4/11/2017 8:38:59 AME0411 13:38:59.685295       1 repair.go:149] the cluster IP 10.43.52.226 for service monitoring-influxdb/kube-system is not allocated; repairing
4/11/2017 8:38:59 AME0411 13:38:59.686447       1 repair.go:186] the cluster IP 10.43.78.151 appears to have leaked: cleaning up
4/11/2017 8:38:59 AME0411 13:38:59.687563       1 repair.go:186] the cluster IP 10.43.181.237 appears to have leaked: cleaning up
4/11/2017 8:38:59 AME0411 13:38:59.688731       1 repair.go:186] the cluster IP 10.43.182.132 appears to have leaked: cleaning up
4/11/2017 8:38:59 AME0411 13:38:59.689859       1 repair.go:186] the cluster IP 10.43.227.178 appears to have leaked: cleaning up
4/11/2017 8:39:00 AME0411 13:39:00.606515       1 errors.go:63] apiserver received an error that is not an unversioned.Status: client: etcd cluster is unavailable or misconfigured
4/11/2017 8:39:00 AMI0411 13:39:00.606983       1 trace.go:61] Trace "Update /api/v1/namespaces/kube-system/endpoints/kube-scheduler" (started 2017-04-11 13:38:45.601876511 +0000 UTC):
4/11/2017 8:39:00 AM[28.441µs] [28.441µs] About to convert to expected version
4/11/2017 8:39:00 AM[74.833µs] [46.392µs] Conversion done
4/11/2017 8:39:00 AM[81.67µs] [6.837µs] About to store object in database
4/11/2017 8:39:00 AM[15.005050214s] [15.004968544s] END
4/11/2017 8:39:04 AME0411 13:39:04.863560       1 errors.go:63] apiserver received an error that is not an unversioned.Status: client: etcd cluster is unavailable or misconfigured
4/11/2017 8:39:04 AMI0411 13:39:04.864028       1 trace.go:61] Trace "Update /api/v1/namespaces/kube-system/endpoints/kube-controller-manager" (started 2017-04-11 13:38:49.859797067 +0000 UTC):
4/11/2017 8:39:04 AM[32.185µs] [32.185µs] About to convert to expected version
4/11/2017 8:39:04 AM[109.287µs] [77.102µs] Conversion done
4/11/2017 8:39:04 AM[113.475µs] [4.188µs] About to store object in database
4/11/2017 8:39:04 AM[15.004159285s] [15.00404581s] END
4/11/2017 8:39:14 AME0411 13:39:14.696507       1 repair.go:77] unable to persist the updated service IP allocations: client: etcd cluster is unavailable or misconfigured
4/11/2017 8:39:19 AME0411 13:39:19.407210       1 errors.go:63] apiserver received an error that is not an unversioned.Status: client: etcd cluster is unavailable or misconfigured
4/11/2017 8:39:19 AMI0411 13:39:19.407411       1 trace.go:61] Trace "Update /api/v1/namespaces/kube-system/endpoints/kube-scheduler" (started 2017-04-11 13:39:04.40236314 +0000 UTC):
4/11/2017 8:39:19 AM[16.871µs] [16.871µs] About to convert to expected version
4/11/2017 8:39:19 AM[45.066µs] [28.195µs] Conversion done
4/11/2017 8:39:19 AM[49.055µs] [3.989µs] About to store object in database
4/11/2017 8:39:19 AM[15.005021227s] [15.004972172s] END
```

**Comment by harutyundermenjyan** [comment]
Hi,
I've tried the plane isolation on the rancher v1.5.6 version and have the same behavior described above.
added the labels to the appropriate hosts that is 3 etcd=true, 2 orchestration=true and 1 compute=true
and after finishing the installation the kubectl shows that there are no any nodes available and all pods in pending state

**Comment by pulberg** [comment]
testing seems better in v1.5.6, if the network is down, it is correctly reported from kubectl.  I still have the issue where kube-system pods do not get moved on network down events.

**Comment by harutyundermenjyan** [comment]
it seems I've find the reason of the issue.
After moving all nodes in the same type of OS the issue is solved.

**Comment by leodotcloud** [comment]
@pulberg we have made a lot of networking fixes since v1.5.3. Can you please try the latest releases and open a new issue if you are still facing problems?

**Comment by verenion** [comment]
I'm seeing the exact issue in v1.6.14, no kubernetes dashboard after setting it up with resilient planes, all fresh hardware and OSs. We are running rancher with an external DB in HA mode, all nodes are reporting as SchedulingDisabled.
